spec/controllers/api/v2/searches_controller_spec.rb

```diff
@@ -276,45 +273,43 @@
     end
 
     context 'when the search options are valid' do
-      let!(:search) { double(ApiI14ySearch, as_json: { foo: 'bar'}, modules: %w(I14Y)) }
+      let!(:search) { instance_double(ApiI14ySearch, as_json: { foo: 'bar' }, modules: %w[I14Y]) }
 
       before do
-        expect(Affiliate).to receive(:find_by_name).and_return(affiliate)
-
-        expect(ApiI14ySearch).to receive(:new).with(hash_including(query: 'api')).and_return(search)
-        expect(search).to receive(:run)
-        expect(SearchImpression).to receive(:log).with(search,
-                                                   'i14y',
-                                                   hash_including('query'),
-                                                   be_a_kind_of(ActionDispatch::Request))
+        allow(Affiliate).to receive(:find_by_name).and_return(affiliate)
+        allow(ApiI14ySearch).to receive(:new).with(hash_including(query: 'api')).and_return(search)
+        allow(search).to receive(:run)
+        allow(SearchImpression).to receive(:log).with(search,
+                                                      'i14y',
+                                                      hash_including('query'),
+                                                      be_a_kind_of(ActionDispatch::Request))
 
-        get :i14y, params: search_params
+        get :i14y, params: search_params.merge(tags: 'tag from params')
       end
 
       it { is_expected.to respond_with :success }
 
       it 'passes the correct options to its ApiI4ySearch object' do
```

Comment:
Primary change in the file... many/most other changes are to appease rubocop while I'm in here.